test/suites/ekuiper/streams_rules_test.go

```diff
@@ -58,10 +71,86 @@ func TestStreamsAndRules(t *testing.T) {
 	utils.TestDeviceVirtualReading(t)
 
 	t.Run("check rule_log", func(t *testing.T) {
-		utils.Exec(t, `edgex-ekuiper.kuiper-cli getstatus rule rule_log`)
+		var ruleStatus RuleStatus
+		var body string
+
+		// waiting for readings to come from edgex to ekuiper
+		for i := 1; ; i++ {
+			time.Sleep(1 * time.Second)
+			stdout, _, err := utils.Exec(t, "edgex-ekuiper.kuiper-cli getstatus rule rule_log")
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			startIndex := strings.Index(stdout, "{")
+
+			if startIndex < 0 {
+				t.Fatal("Error getting status of rule rule_log in JSON format")
+			} else {
+				body = stdout[startIndex:]
+			}
+
+			err = json.Unmarshal([]byte(body), &ruleStatus)
```

Comment:
I now understand why you said it was difficult. Why don't you just call the REST API with http.Get?

```
$ curl http://localhost:59720/rules/rule_log/status
{
  "source_stream1_0_records_in_total": 0,
  "source_stream1_0_records_out_total": 0,
  "source_stream1_0_exceptions_total": 0,
  "source_stream1_0_process_latency_us": 0,
  "source_stream1_0_buffer_length": 0,
  "source_stream1_0_last_invocation": 0,
  "op_2_filter_0_records_in_total": 0,
  "op_2_filter_0_records_out_total": 0,
  "op_2_filter_0_exceptions_total": 0,
  "op_2_filter_0_process_latency_us": 0,
  "op_2_filter_0_buffer_length": 0,
  "op_2_filter_0_last_invocation": 0,
  "op_3_project_0_records_in_total": 0,
  "op_3_project_0_records_out_total": 0,
  "op_3_project_0_exceptions_total": 0,
  "op_3_project_0_process_latency_us": 0,
  "op_3_project_0_buffer_length": 0,
  "op_3_project_0_last_invocation": 0,
  "sink_log_0_0_records_in_total": 0,
  "sink_log_0_0_records_out_total": 0,
  "sink_log_0_0_exceptions_total": 0,
  "sink_log_0_0_process_latency_us": 0,
  "sink_log_0_0_buffer_length": 0,
  "sink_log_0_0_last_invocation": 0
}
```

Reply:
Oh, I didn't aware that the results are different between kuiper CLI and REST API. Updated!

Reply:
I think this is really a bug on the CLI. Only the JSON should be printed to the standard output and everything else should go to standard error to allow capturing the payload.

Reply:
Agree. I think `Connecting to 127.0.0.1:20498...` should not be in stdout or stderr at all. maybe it should be in debugging log.
Would you like to open an issue in ekuiper repo?